vignettes/layer.Rmd

```diff
@@ -16,4 +16,301 @@ knitr::opts_chunk$set(
+
+First, we know that the cells should be consistently formatted as follows, which allows us to create a body_plan:
+
+  - Total counts to be whole numbers 
+  - Cells under each treatment to be a combination of a count and percent of that population. When percent is 100 or zero, value should be blank.
+  - p-values to be formatted where: >.99 is displayed as ">0.99", <0.001 is displayed as "<0.001", and any values between should be displayed to three decimal places. Missing p-values should be replaced with "--".
+  
+Next, we can apply standard to our organizations AE ARD format to ensure that names of are set consistently:
+
+  - `AEBODYSYS` defines the body systems and overall grouping of the data.
+  - `AETERM` defines the labels to display in the table.
+  - `value` defines the values to present in the table.
+  - `treatment` defines the treatments and column for each value.
+  - `param` is the column that describes the params of the `value` column. "AEs" indicates that the number in `value` represents the total counts, "n" represents the count for the specific AE, "pct" is the percent of the treatment population, and finally "pval" is the p-value.
+  
+With this information we can construct a template AE `tfrmt`:
+  
+```{r}
+
+ae_tfrmt_template <- tfrmt(
+    group = AEBODSYS,
+    label = AETERM,
+    param = param,
+    column = treatment,
+    value = value,
+  body_plan = body_plan(
+    ## All entries where the param column is `AEs` (representing total counts)
+    frmt_structure(
+      group_val = ".default",
+      label_val = ".default",
+      AEs = frmt("[XXX]")
+      ),
+    
+    ## Combine entries where param column is `n` and `pct` to create a cell for
+    ## that population
+    frmt_structure(
+      group_val = ".default",
+      label_val = ".default",
+      frmt_combine(
+        "{n} {pct}",
+        n = frmt("XXX"),
+        pct = frmt_when(
+          "==100" ~ "",
+          "==0" ~ "",
+          TRUE ~ frmt("(xx.x %)")
+          )
+        )
+      ),
+    
+    ## All entries where param column is `pval`, format conditionally.
+    ## When the value is missing, replace the NA with "--".
+    frmt_structure(
+       group_val = ".default", 
+       label_val = ".default",
+       pval = frmt_when(
+         ">0.99" ~ ">0.99",
+         "<0.001" ~ "<0.001",
+         TRUE ~ frmt("x.xxx", missing = "--")
+         )
+    )
+  )
+)
+
+
+```
+
+## Piping - two layer tfrmts
+
+Now that we have a template to work off of, we can create a `tfrmt` specifically for the table we are creating for our study. This can be done a few ways. The first method that can be used is to pipe another tfrmt. The first argument of `trfmt` is "tfrmt_obj". This allows for `tfrmt` to be layered directly using a pipe. 
```

Comment:
```suggestion
Now we have a template to serve as the basis for our AE tables, which reduces much of the overhead in our formatting. We can continue to build on this template by making a study-specific `tfrmt` to be layered on top. This can be done a few ways. The first method that can be used is to pipe another tfrmt. The first argument of `trfmt` is "tfrmt_obj". This allows for `tfrmt` to be layered directly using a pipe. 
```